js/components/meta-select-control.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * WordPress dependencies.
+ */
+const { __ } = wp.i18n;
+const { SelectControl } = wp.components;
+const { withSelect, withDispatch } = wp.data;
+const { compose } = wp.compose;
+
+const MetaSelectControl = compose(
+	withDispatch( function( dispatch, props ) {
+		return {
+			setMetaValue( value ) {
+				dispatch( 'core/editor' ).editPost( {
+					meta: { [ props.metaKey ]: value },
+				} );
+			},
+		};
+	} ),
+	withSelect( function( select, props ) {
+		const currentPostId = select( 'core/editor' ).getCurrentPostId();
+		const query = {
+			per_page: -1,
+			exclude: currentPostId,
+			metaKey: 'sub_menu',
+			metaValue: 0,
+		};
```

Comment:
This is great, except that it only includes menu items registered by Hello Admin. What if I want to make my page a sub-menu of Dashboard, or Appearance?

So instead of doing this standard `getEntityRecords` REST API call, we could [register a custom Rest API endpoint](https://developer.wordpress.org/rest-api/extending-the-rest-api/adding-custom-endpoints/) and use `[apiFetch](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-api-fetch/)` to retrieve the results.

From the custom endpoint, just grab the `$menu` global, which should contain all the registered primary menu items, and filter out things like separators, maybe simplify it a little. Then return that!

Reply:
This is an awesome idea and fully agree with you. However, I am having a problem getting the global $menu. I have registered the route and it is working correctly but it always returns as null. I have done some research and found that it is not available at this action hook and can't find a workaround.
https://wordpress.stackexchange.com/questions/396528/cannot-get-global-menu-value-in-form-request

Any ideas?

`add_action( 'rest_api_init', [ $this, 'register_custom_routes' ], 100 );

public function register_custom_routes() {
  register_rest_route(
	  'hello-admin/v1',
	  '/menus',
	  [
		  'methods'  => 'GET',
		  'callback' => [ $this, 'return_all_registered_menus' ],
	  ]
  );
}

public function return_all_registered_menus() {
  global $menu;
  
  return wp_json_encode( $menu );
}`

Reply:
What do you think about what I've done to solve this in https://github.com/hello-wp/hello-admin/pull/6/commits/440b35d3af4ed0827e6e613d6eb3de6efcfe132f?

Reply:
This is awesome having access to the $menu global in the transient! Call the genius police!